notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+
+
+Vectors can be added and scaled element-wise:
+
+$$a \vec{x} + b \vec{y} = \begin{bmatrix} a \ x_1 & + & b \  y_1 \\
+                                            & \vdots & \\
+                                          a \  x_n & + & b \  y_n
+                          \end{bmatrix}
+$$
+
+#### The $\vec{1}$ Vector
+In any $d$ dimensional space, $\vec{1}$ is the vector of all $1$'s: 
+$ \begin{bmatrix} 1 \\ \vdots \\ 1  \end{bmatrix}$
+
+#### Span
+The span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$ is the set of all possible linear combinations. For these $p$ vectors:
+
+$$ \{ c_1 \ \vec{v_1} + c_2 \ \vec{v_2} + \dots + c_p \ \vec{v_p} \ : \ \forall c_i \in F\}$$
+
+where $F$ is the field of the vector space (out of scope).
+
+#### Vector Spaces
+A vector space $V$ is the span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$, where each $\vec{v_i}$ is a $n \times 1$ dimensional column vector.
+
+#### Vector Subspaces
+A subspace $U$ of $V$ is the span of a set of vectors $\{\vec{u_1}, \dots, \vec{u_q}\}$ where each $\vec{u_i} \in V$. This means every vector in $U$ is also in $V$.
+
+#### Vector Angles
+When you put any two vectors terminal end to terminal end without changing their direction, you can measure the angle between them.
+
+<img src="../../notebooks-images/vector_space_review__angle.png" width="300" />
+
+
+#### Vector Length
+
+Intuition in $\mathbb{R}^2$ :
+
+Recall the triangle method of adding two vectors. If we add two perpendicular vectors $\vec{a} + \vec{b}$ in $\mathbb{R}^2$, then we know that the resulting vector will be the hypotenuse. In this case, we also know that the length of $\vec{a} + \vec{b}$ will follow the Pythagorean Theorem: $\sqrt{a^2 + b^2}$.
+
+<img src="../../notebooks-images/vector_space_review__length.png" width="400" />
+
+
+General Formula for  Length of $\vec{v} \in \mathbb{R}^n$ :
+$$\begin{aligned} || \vec{v} || \quad
+&=  \quad \sqrt{v_1^2 + v_2^2 + \dots + v_n^2}  \\
+&= \quad \sqrt{\vec{v} \cdot \vec{v}}
+\end{aligned}
+$$
+
+Where the final operator is the dot product.
+
+$$ \begin{aligned}
+\vec{x} \cdot \vec{y} \quad 
+&= \quad x_1 \ y_1 + x_2 \ y_2 + \dots + x_n \ y_n \\
+&= \quad||u|| \ ||v|| \ \cos{\theta}
+\end{aligned}
+$$
+
+The first expression is known as the algebraic definition of the dot product, and the second is the geometric definition. Note that the dot product is the inner product defined for vectors in $\mathbb{R}^n$. 
+
+#### Distance Between Two Vectors
+$$dist(\vec{x},\vec{y}) \quad = \quad || \vec{x} - \vec{y} ||$$
+
+<img src="../../notebooks-images/vector_space_review__distance.png" width="500" />
+
+
+#### Orthogonal Vectors
+For two non-zero vectors to be orthogonal, they must satisfy the property that $\vec{x} \cdot \vec{y} = 0$. Since they have non-zero length, the only way for the two vectors to be orthogonal is if $\cos{\theta} = 0$. One satisfying $\theta$ is 90 degrees, our familiar right angle.
```

Comment:
HAHAHAHA, satisfying.. you are so cute! :)

Reply:
wait why is that cute/funny?

Reply:
just the choice of words :P